soc/nordic/soc.yml

```diff
@@ -1,39 +1,73 @@
 family:
-- name: nordic_nrf
-  series:
-  - name: nrf51
-    socs:
-    - name: nrf51822
-  - name: nrf52
-    socs:
-    - name: nrf52805
-    - name: nrf52810
-    - name: nrf52811
-    - name: nrf52820
-    - name: nrf52832
-    - name: nrf52833
-    - name: nrf52840
-  - name: nrf53
-    socs:
-    - name: nrf5340
-      cpuclusters:
-      - name: cpuapp
-      - name: cpunet
-  - name: nrf54l
-    socs:
-    - name: nrf54l15
-      cpuclusters:
-      - name: cpuapp
-  - name: nrf54h
-    socs:
-    - name: nrf54h20
-      cpuclusters:
-      - name: cpuapp
-      - name: cpurad
-      - name: cpuppr
-  - name: nrf91
-    socs:
-    - name: nrf9131
-    - name: nrf9151
-    - name: nrf9160
-    - name: nrf9161
+  - name: nordic_nrf
+    series:
+      - name: nrf51
+        socs:
+          - name: nrf51822
+      - name: nrf52
+        socs:
+          - name: nrf52805
+          - name: nrf52810
+          - name: nrf52811
+          - name: nrf52820
+          - name: nrf52832
+          - name: nrf52833
+          - name: nrf52840
+      - name: nrf53
+        socs:
+          - name: nrf5340
+            cpuclusters:
+              - name: cpuapp
+              - name: cpunet
+      - name: nrf54l
+        socs:
+          - name: nrf54l15
+            cpuclusters:
+              - name: cpuapp
+      - name: nrf54h
+        socs:
+          - name: nrf54h20
+            cpuclusters:
+              - name: cpuapp
+              - name: cpurad
+              - name: cpuppr
+      - name: nrf91
+        socs:
+          - name: nrf9131
+          - name: nrf9151
+          - name: nrf9160
+          - name: nrf9161
+
+# Recovery/erase is only needed once per core.
+run_once:
```

Comment:
a top-level "run_conce" key feels weird here, I'd suggest something like:

```
- flashing:
  - singlerun:
  ...
  - anyfuturestuff:  
```

Reply:
(or `flash.yml`...)

Reply:
This was a separate file initially but moved into these files because "of too many yaml files" concerns raised by others

Reply:
> This was a separate file initially but moved into these files because "of too many yaml files" concerns raised by others

IMHO mixing files purpose is not a good idea either.

Reply:
> IMHO mixing files purpose is not a good idea either.

feel free to join the discussion: https://github.com/zephyrproject-rtos/zephyr/pull/50305#discussion_r1400797335
I feel the same, multiple files can be easier to read understand if the are well named and organized, quote:
> The fact that both files are in yaml doesn't mean there should be one huge yaml file containing everything.

But there was request to unify and have a single source for various meta data:
https://github.com/zephyrproject-rtos/zephyr/pull/50305#discussion_r1407873481